source/server/listener_manager_impl.cc

```diff
@@ -126,9 +128,10 @@ ListenerImpl::ListenerImpl(const envoy::api::v2::Listener& config, const std::st
       workers_started_(workers_started), hash_(hash),
       local_drain_manager_(parent.factory_.createDrainManager(config.drain_type())),
       config_(config), version_info_(version_info) {
-  // TODO(htuch): Support multiple filter chains #1280, add constraint to ensure we have at least on
-  // filter chain #1308.
-  ASSERT(config.filter_chains().size() >= 1);
+  if (config.filter_chains().empty()) {
```

Comment:
Can we do this via proto constraint? Do we need the check here?

Reply:
Actually, it looks that it's already enforced by the proto constraint.

I removed the check and test for it.